tests/generator-Tests/Unit-Tests/CodeGeneratorExpectedResults/Common/WriteDuplicateInterfaceEventArgs.txt

```diff
@@ -70,15 +70,25 @@ internal partial class AnimatorListenerInvoker : global::Java.Lang.Object, Anima
 #pragma warning disable 0169
 	static Delegate GetOnAnimationEnd_IHandler ()
 	{
-		if (cb_OnAnimationEnd_OnAnimationEnd_I_Z == null)
-			cb_OnAnimationEnd_OnAnimationEnd_I_Z = JNINativeWrapper.CreateDelegate (new _JniMarshal_PPI_Z (n_OnAnimationEnd_I));
-		return cb_OnAnimationEnd_OnAnimationEnd_I_Z;
+		return cb_OnAnimationEnd_OnAnimationEnd_I_Z ??= new _JniMarshal_PPI_Z (n_OnAnimationEnd_I);
 	}
 
+	[global::System.Diagnostics.DebuggerDisableUserUnhandledExceptions]
 	static bool n_OnAnimationEnd_I (IntPtr jnienv, IntPtr native__this, int param1)
 	{
-		var __this = global::Java.Lang.Object.GetObject<java.code.AnimatorListener> (jnienv, native__this, JniHandleOwnership.DoNotTransfer);
-		return __this.OnAnimationEnd (param1);
+		var __envp = new global::Java.Interop.JniTransition (jnienv);
+		var __r = global::Java.Interop.JniEnvironment.Runtime;
```

Comment:
I should have called this out *more* than just in the draft commit message, but the problem with this is that `JniEnvironment.Runtime` can throw:

https://github.com/dotnet/java-interop/blob/14f94a57643d3bbdd7b45b75b333b16bbead0e30/src/Java.Interop/Java.Interop/JniEnvironment.cs#L221

It *shouldn't* throw, ideally, but it *could*, if e.g. the runtime hasn't been initialized yet, or if the runtime has been shutdown.

If it throws an exception outside of the try/catch, we can unwind past the Java frames, and corrupt the JVM.

We could ignore this possibility, and keep this as-is, or we can go with what the draft message suggested:

```csharp
var __envp = new global::Java.Interop.JniTransition (jnienv);
global::Java.Interop.JniRuntime? __r = null;
try {
  __r = global::Java.Interop.JniEnvironment.Runtime;
} …
```

and *alternate* approach would be to introduce more public methods:

```csharp
namespace Java.Interop;

partial class JniEnvironment {
    public static bool BeginMarshalMethod (IntPtr jnienv, out JniTransition envp, [NotNullWhen(true)] out JniRuntime? runtime);
    public static void EndMarshalMethod (ref JniTransition envp);
}
```

allowing:

```csharp
if (!JniEnvironment.BeginMarshalMethod (jnienv, out var __envp, out var __r))
    return /* default */;
try {
    __r.OnEnterMarshalMethod ();
    // …
} catch (Exception e) {
    __r.OnUserUnandledException (ref __e, e);
} finally {
    JniEnvironment.EndMarshalMethod (ref __envp);
}
```

---

A question that comes to mind is IL size: which is smaller, between:
  *  "just call `JniEnvironment.Runtime`"
  * Call `JniEnvironment.Runtime` within the try block
  * JniEnvironment.BeginMarshalMethod()

Enter my sample app (see below).

  *  A(): "just call `JniEnvironment.Runtime`"
    Code size 44
  * B(): Call `JniEnvironment.Runtime` within the try block
    Code size: 49
  * D(): JniEnvironment.BeginMarshalMethod()
    Code size: 43

which makes it look like `JniEnvironment.BeginMarshalMethod()` + `JniEnvironment.EndMarshalMethod()` has the smallest impact on code size, while still permitting correctness.

```csharp
// See https://aka.ms/new-console-template for more information
Console.WriteLine("Hello, World!");

void A ()
{
    var __e = new JniTransition (0);
    var r = JniEnvironment.Runtime;
    try {
        r.OnEnterMarshalMethod();
    } catch (Exception e) {
        r.OnUserUnandledException (ref __e, e);
    } finally {
        __e.Dispose();
    }
}

void B ()
{
    var __e = new JniTransition (0);
    JniRuntime? r = null;
    try {
        r = JniEnvironment.Runtime;
        r.OnEnterMarshalMethod();
    } catch (Exception e) {
        r?.OnUserUnandledException (ref __e, e);
    } finally {
        __e.Dispose();
    }
}

void C ()
{
    if (!JniEnvironment.BeginMarshalMethod (0, out var __e, out var r))
        return;
    try {
        r.OnEnterMarshalMethod();
    } catch (Exception e) {
        r.OnUserUnandledException (ref __e, e);
    } finally {
        __e.Dispose();
    }
}

void D ()
{
    if (!JniEnvironment.BeginMarshalMethod (0, out var __e, out var r))
        return;
    try {
        r.OnEnterMarshalMethod();
    } catch (Exception e) {
        r.OnUserUnandledException (ref __e, e);
    } finally {
        JniEnvironment.EndMarshalMethod (ref __e);
    }
}

struct JniTransition {
    public JniTransition (IntPtr env) {}
    public void Dispose() {}
}

class JniEnvironment {
    public static JniRuntime Runtime => throw null;
    public static bool BeginMarshalMethod (int env, out JniTransition __e, out JniRuntime r) => throw null;
    public static void EndMarshalMethod (ref JniTransition __e) => throw null;
}

class JniRuntime {
    public class JniValueManager {

    }

    public virtual void OnUserUnandledException (ref JniTransition env, Exception e)
    {
    }

    public virtual void OnEnterMarshalMethod ()
    {
    }

    public JniValueManager ValueManager => throw null!;
}
```